assets/src/domain/eventEditor/ui/tickets/useEditTicketModal.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback } from 'react';
+import { __ } from '@wordpress/i18n';
+import { FormProps } from 'react-final-form';
+
+import useTicketFormConfig from './ticketForm/useTicketFormConfig';
+import { useEditorModal, EditorModal, ModalClose } from '@appLayout/editorModal';
+import { useTicketMutator, UpdateTicketInput } from '@edtrServices/apollo/mutations';
+import useEvent from '@edtrServices/apollo/queries/events/useEvent';
+
+const useEditTicketModal: EditorModal = (entityId) => {
+	const { updateEntity } = useTicketMutator(entityId);
+	const { closeEditor } = useEditorModal();
+
+	const { name: eventName = '' } = useEvent() || {};
+
+	const onClose = useCallback<ModalClose>((): void => {
```

Comment:
Same here, can we avoid using `useCallback` here?

Reply:
It has a dependency `closeEditor`

Reply:
the dependency doesn't matter at all, he's suggesting we simply return a callback that is not memoized, ie:

```js
const onClose = (): void => {
    closeEditor('editTicket');
};
```

but the problem with that is a new function will be created every time this hook is called, which could easily cause unnecessary re-renders in components that consume this callback. Since this is ultimately using a provider to get the required logic, I think it's fair to use memoization here.

Reply:
Great explanation, @tn3rb, thx.
My understanding is that this `useCallback` is preventing re-renders from a different cause/source.
To me, it looks weird to apply memoization to a close handler or any other similar handler which is just dispatching an action.

Reply:
We need to pinpoint the exact source of re-renders.
Applying `useCallback` in cases where it is not required makes it hard to reason about the code.

Reply:
> My understanding is that this useCallback is preventing re-renders from a different cause/source.

Not sure I totally grok what you are saying...

It's not preventing from the source, ie: where `closeEditor()` is coming from, it's preventing re-renders "downstream (ie: consumers of this hook). In other words, any components that receive the `onClose()` callback will unnecessarily re-render every time this hook is called, unless  `onClose()` is memoized. Sorry if I'm repeating something you already totally get.

> We need to pinpoint the exact source of re-renders.

I agree, and normally a component will only render when its props change. The problem with passing functions as props is that they will ALWAYS be considered changed because of the way that props are compared, even if the function itself has not undergone any changes. 

For example, in the following totally contrived example, a function is passed as a prop to another component:

```js
const ComponentA = () => {
    const getBobsName = () => 'Bob';
    return (
        <ComponentB callback={ getBobsName } />
    );
}
```

`ComponentB` will be re-rendered everytime `ComponentA` renders because it will see the incoming `getBobsName` function as being different than the one it was passed previously, even though there is no possible way for that function to be anything different. Memoizing that callback however would prevent `ComponentB` from re-rendering. In the case of a callback as simple as this, it's probably more expensive to memoize the result than it is to simply rerun the function, so memoization here would be unnecessary. The callback that Manzoor memoized is pulling data from a Provider, so one would think that is more expensive of an operation and therefore memoization is likely useful.

So anytime a callback of any kind of higher complexity is being passed as a prop, it's likely wise to just memoize it.

Now that said, in this case, since we are dealing with a hook, we could check if the callback is actually being passed as a prop to another component. If it is NOT, and the callback is being utilized immediately within the consuming component, then the memoization is unnecessary (assuming the hook is not used elsewhere). But without even looking, we know that this is not the case here, because the  `onClose()` and `onSubmit()` callbacks are most certainly being passed along to another component to be used as actions that are triggered by form events.

So bottom line:  any callback that is being passed as a prop to another component should be memoized, **UNLESS** that callback is so painfully simple that memoization would likely decrease performance.

But I totally agree with your comments with regards to other forms of memoization,  I also appreciate your vigilance with regards to maintaining and the readability of code 👍 So plz do not hesitate to comment on things like this again in the future, as it is always good to be reminded about these things.